test-runner/chrome/content/loggest-chrome-runner.js

```diff
@@ -1016,21 +952,38 @@ function _runTestFile(runner, testFileName, variant, baseUrl, manifestUrl,
 
     if (event.data.type === 'loggest-test-results') {
       console.harness('calling writeTestLog and resolving');
-      var jsonStr = event.data.data,
-          logData = JSON.parse(jsonStr);
-      // this must be done prior to the compartment getting killed
-      var summary = summaryFromLoggest(testFileName, variant, logData);
-      if (!TEST_PARAMS.logFailuresOnly || summary.result === 'fail') {
-        writeTestLog(testFileName, variant, jsonStr, summary).then(
-          function() {
-            console.harness('write completed!');
-            deferred.resolve(summary);
-          });
-      }
-      else {
-        console.harness('not a failure, not writing');
-        deferred.resolve(summary);
-      }
+      var testLogs = JSON.parse(event.data.data);
+      var time = Date.now();
+      var basename = testFileName + '-' + variant.replace(/:/g, '_');
+
+      var summary = {
+        filename: testFileName,
+        timestamp: testRunTimestampMS,
+        href: 'test-logs/' + basename + '-' + time + '.json',
+        tests: testLogs.map((resultData) => {
+          return { name: resultData.name,
+                   variant: resultData.variant,
+                   lastError: resultData.lastError,
+                   result: resultData.result };
+        }),
+        result: testLogs.every((resultData) => resultData.result === 'pass') ?
+              'pass' : 'fail'
+      };
+
+      var logResults = {
+        type: 'suite-results',
+        filename: summary.filename,
+        href: summary.href,
+        timestamp: time,
+        result: summary.result,
+        tests: testLogs
+      };
+
+      writeFile('.', summary.href,
+                'document.currentScript.resolve(' + JSON.stringify(logResults) + ')')
```

Comment:
Do care: I see that in conjunction with "fetch" in logic-inspector/js/main.js this seems to be a JSONP-style method of loading JSON, but with Promises.  I hate the XHR boilerplate too, but luckily I have made a shoddy one we can use at https://github.com/asutherland/glodastrophe/blob/master/www/app/fetch.js.

Note that while from a security/hygiene perspective, JSONP is bad, a potentially bigger concern is just that it makes it hard to use other tools to process the JSON.